### PR TITLE
Fuchsia gni and material font

### PR DESCRIPTION
**Pull request by xster**
I have no idea if this is right. I basically copied this change into fuchsia/flutter/build/flutter_app.gni then went to fuchsia/magenta/script and built all. Don't know if it's the right way to test.

For https://github.com/flutter/flutter/issues/6670

### CODE REVIEW

File: build/flutter_app.gni [hunk 1]
```diff
@@ -132,6 +133,8 @@ template("flutter_app") {
     ]
     if (defined(invoker.manifest)) {
       inputs += [ rebase_path(invoker.manifest) ]
+    } else {
+      inputs += [ rebase_path(flutter_tools_asset_manifest)]
     }
```
Comment (abarth) [inline build/flutter_app.gni]: We should set a local variable called `manifest` that is either `invoker.manifest` (if that is defined) or `flutter_tools_asset_manifest`.  Then we can add it to `inputs` and construct the command line below without conditionals.

File: build/flutter_app.gni [hunk 2]
```diff
@@ -72,6 +72,7 @@ template("flutter_app") {
   flutter_tools_name = get_label_info(flutter_tools_label, "name")
   flutter_tools_packages = "$flutter_tools_gen_dir/$flutter_tools_name.packages"
   flutter_tools_main = "$flutter_tools_label/bin/fuchsia_builder.dart"
+  flutter_tools_asset_manifest = "$flutter_tools_label/schema/material_fonts.yaml"
```
Comment (abarth) [inline build/flutter_app.gni]: This file isn't the correct default.  Instead, we need to construct a default `pubspec.yaml` and use that as the default.

**Comment by xster** [comment]
It produces a flx file structure of 

Archive:  email_story.flx
warning [email_story.flx]:  45 extra bytes at beginning or within zipfile
  (attempting to process anyway)
warning:  skipped "../" path component(s) in ../../apps/fonts/third_party/roboto/Roboto-Regular.ttf
  inflating: apps/fonts/third_party/roboto/Roboto-Regular.ttf
warning:  skipped "../" path component(s) in ../../apps/fonts/third_party/roboto/Roboto-Italic.ttf
  inflating: apps/fonts/third_party/roboto/Roboto-Italic.ttf
warning:  skipped "../" path component(s) in ../../apps/fonts/third_party/roboto/Roboto-Thin.ttf
  inflating: apps/fonts/third_party/roboto/Roboto-Thin.ttf
warning:  skipped "../" path component(s) in ../../apps/fonts/third_party/roboto/Roboto-ThinItalic.ttf
  inflating: apps/fonts/third_party/roboto/Roboto-ThinItalic.ttf
warning:  skipped "../" path component(s) in ../../apps/fonts/third_party/roboto/Roboto-Light.ttf
  inflating: apps/fonts/third_party/roboto/Roboto-Light.ttf
warning:  skipped "../" path component(s) in ../../apps/fonts/third_party/roboto/Roboto-LightItalic.ttf
 extracting: apps/fonts/third_party/roboto/Roboto-LightItalic.ttf
warning:  skipped "../" path component(s) in ../../apps/fonts/third_party/roboto/Roboto-Medium.ttf
  inflating: apps/fonts/third_party/roboto/Roboto-Medium.ttf
warning:  skipped "../" path component(s) in ../../apps/fonts/third_party/roboto/Roboto-MediumItalic.ttf
 extracting: apps/fonts/third_party/roboto/Roboto-MediumItalic.ttf
warning:  skipped "../" path component(s) in ../../apps/fonts/third_party/roboto/Roboto-Bold.ttf
 extracting: apps/fonts/third_party/roboto/Roboto-Bold.ttf
warning:  skipped "../" path component(s) in ../../apps/fonts/third_party/roboto/Roboto-BoldItalic.ttf
 extracting: apps/fonts/third_party/roboto/Roboto-BoldItalic.ttf
warning:  skipped "../" path component(s) in ../../apps/fonts/third_party/roboto/Roboto-Black.ttf
 extracting: apps/fonts/third_party/roboto/Roboto-Black.ttf
warning:  skipped "../" path component(s) in ../../apps/fonts/third_party/roboto/Roboto-BlackItalic.ttf
 extracting: apps/fonts/third_party/roboto/Roboto-BlackItalic.ttf
warning:  skipped "../" path component(s) in ../../apps/fonts/third_party/robotocondensed/RobotoCondensed-Regular.ttf
 extracting: apps/fonts/third_party/robotocondensed/RobotoCondensed-Regular.ttf
warning:  skipped "../" path component(s) in ../../apps/fonts/third_party/robotocondensed/RobotoCondensed-Italic.ttf
  inflating: apps/fonts/third_party/robotocondensed/RobotoCondensed-Italic.ttf
warning:  skipped "../" path component(s) in ../../apps/fonts/third_party/robotocondensed/RobotoCondensed-Light.ttf
 extracting: apps/fonts/third_party/robotocondensed/RobotoCondensed-Light.ttf
warning:  skipped "../" path component(s) in ../../apps/fonts/third_party/robotocondensed/RobotoCondensed-LightItalic.ttf
 extracting: apps/fonts/third_party/robotocondensed/RobotoCondensed-LightItalic.ttf
warning:  skipped "../" path component(s) in ../../apps/fonts/third_party/robotocondensed/RobotoCondensed-Bold.ttf
 extracting: apps/fonts/third_party/robotocondensed/RobotoCondensed-Bold.ttf
warning:  skipped "../" path component(s) in ../../apps/fonts/third_party/robotocondensed/RobotoCondensed-BoldItalic.ttf
 extracting: apps/fonts/third_party/robotocondensed/RobotoCondensed-BoldItalic.ttf
  inflating: AssetManifest.json
  inflating: FontManifest.json
  inflating: LICENSE
  inflating: snapshot_blob.bin

Not sure if the paths are correct

**Comment by abarth** [comment]
> Not sure if the paths are correct

We might want to clean up the logic in flutter_tools to strip leading `../..` from these paths.  See https://github.com/flutter/flutter/issues/8403

**Comment by xster** [comment]
The directory structure that it unpacked into from the .flx were in the shape of `apps/fonts/third_party/roboto/Roboto-Regular.ttf`. Is that a correct location or should it be something els?

**Comment by abarth** [comment]
I doesn't really matter what structure is used inside the FLX as long as the font manifest JSON file matches.

**Comment by xster** [comment]
Ah ok. Think I understand. Sounds like using a single relative path string to reference pre-build asset locations and post-build artifact path is causing strangeness. Will take a look at flutter/flutter#8403.

**Comment by xster** [comment]
Closing this since fuchsia no longer needs this manifest

**Pull request closed without merging**